site-src/concepts/guidelines.md

```diff
@@ -38,6 +38,28 @@ prescribed resolution, the following guiding principles should be applied:
   should be given to the resource appearing first in alphabetical order
   (namespace/name). For example, foo/bar would be given precedence over foo/baz.
 
+## Gracefully Handling Future API Versions
+
+An important consideration when implementing this API is how it might change in
+the future. Similar to the Ingress API before it, this API is designed to be
+implemented by a variety of different products within the same cluster. That
+means that the API version your implementation was developed with may be
+different than the API version it is used with. At a minimum, the following
+requirements must be met to ensure future versions of the API do not break your
+implementation:
+
+* Handle fields with loosened validation without crashing
```

Comment:
We should be more specific here:

"Fields values may contain values with less restrictive validation in newer versions of the API. In this case, the controller is expected to handle this gracefully. The most common response would be to reject the configuration as malformed and signal the user via a condition in the `status` block.

Reply:
Updated, thanks!